File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/ControlPanelParam.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel;

import android.databinding.BaseObservable;
import android.databinding.Bindable;
import android.support.annotation.ColorInt;

import net.mm2d.dmsexplorer.BR;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class ControlPanelParam extends BaseObservable {
    private int mBottomPadding;
    private int mMarginRight;
    private int mBackgroundColor;

    @Bindable
    public int getBottomPadding() {
        return mBottomPadding;
    }

    public void setBottomPadding(final int padding) {
        mBottomPadding = padding;
        notifyPropertyChanged(BR.bottomPadding);
    }

    @Bindable
    public int getMarginRight() {
        return mMarginRight;
    }

    public void setMarginRight(final int margin) {
        mMarginRight = margin;
        notifyPropertyChanged(BR.marginRight);
    }

    @Bindable
    public int getBackgroundColor() {
        return mBackgroundColor;
    }

    public void setBackgroundColor(@ColorInt final int color) {
        mBackgroundColor = color;
        notifyPropertyChanged(BR.backgroundColor);
    }
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/ControlPanelParam.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
class ControlPanelParam {
    private val bottomPaddingFlow: MutableStateFlow<Int> = MutableStateFlow(0)
    fun getBottomPaddingFlow(): Flow<Int> = bottomPaddingFlow
    fun setBottomPadding(padding: Int) {
        bottomPaddingFlow.value = padding
    }

    private val marginRightFlow: MutableStateFlow<Int> = MutableStateFlow(0)
    fun getMarginRightFlow(): Flow<Int> = marginRightFlow
    fun setMarginRight(margin: Int) {
        marginRightFlow.value = margin
    }

    private val backgroundColorFlow: MutableStateFlow<Int> = MutableStateFlow(0)
    fun getBackgroundColorFlow(): Flow<Int> = backgroundColorFlow
    fun setBackgroundColor(color: Int) {
        backgroundColorFlow.value = color
    }
}
